knimin/lib/data_access.py

```diff
@@ -3989,14 +4055,37 @@ def read_targeted_plate(self, plate_id):
                         JOIN pm.tm300_8_tool t300 USING (tm300_8_tool_id)
                         JOIN pm.tm50_8_tool t50 USING (tm50_8_tool_id)
                         JOIN pm.water_lot w USING (water_lot_id)
+                        JOIN pm.plate_type pt ON
+                            p.targeted_primer_plate_id = plate_type_id
                      WHERE targeted_plate_id = %s"""
             TRN.add(sql, [plate_id])
             res = TRN.execute_fetchindex()
             if not res:
                 raise ValueError("Target Gene plate %s does not exist"
                                  % plate_id)
             # Magic number 0 -> there is only 1 result row
-            return dict(res[0])
+            res = dict(res[0])
+            rows = res.pop('rows')
+            cols = res.pop('cols')
+
+            # get well values
+            res['raw_concentration'] = None
+            res['mod_concentration'] = None
+            sql = """SELECT row, col, raw_concentration, mod_concentration
+                     FROM pm.targeted_plate_well_values
+                     WHERE targeted_plate_id = %s"""
+            TRN.add(sql, [plate_id])
+            vals = TRN.execute_fetchindex()
+            if vals:
+                res['raw_concentration'] = np.zeros((rows, cols))
+                for (row, col, raw_concentration, mod_concentration) in vals:
+                    res['raw_concentration'][row, col] = raw_concentration
+                    if mod_concentration is not None:
+                        if res['mod_concentration'] is None:
```

Comment:
This is inside a for loop, so it will be None only during the first iteration of the for loop.